game-core/src/main/java/games/strategy/triplea/ai/pro/util/ProTransportUtils.java

```diff
@@ -173,13 +202,16 @@ public static int findUnitsTransportCost(final List<Unit> units) {
         || units.isEmpty()) {
       return Collections.singletonList(unit);
     }
-    units.sort(Comparator.<Unit>comparingInt(u -> TripleAUnit.get(u).getMovementLeft())
-        .thenComparing(getDecreasingAttackComparator(player)));
     final List<Unit> results = new ArrayList<>();
     results.add(unit);
     if (Matches.unitIsLandTransportWithoutCapacity().test(unit)) {
+      units.sort(Comparator.<Unit>comparingInt(u -> TripleAUnit.get(u).getMovementLeft())
+          .thenComparing(getDecreasingAttackComparator(player)));
       results.add(units.get(0));
     } else {
+      units.sort(Comparator.<Unit>comparingInt(u -> TripleAUnit.get(u).getMovementLeft())
+          .thenComparing(Comparator.<Unit>comparingInt(u -> UnitAttachment.get(u.getType()).getTransportCost()))
```

Comment:
This can be simplified to thenComparingInt and then the lambda

Reply:
Done.